knowledge_engine/knowledge_engine/conversion/office_legacy.py

```
@@ -0,0 +1,141 @@
+
+    configured = os.getenv("SOFFICE_PATH") or os.getenv("LIBREOFFICE_PATH")
+    candidates = (configured,) if configured else ()
+    candidates = candidates + _SOFFICE_CANDIDATES
+
+    for candidate in candidates:
+        if not candidate:
+            continue
+
+        resolved = shutil.which(candidate) or candidate
+        try:
+            result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)